src/nvim/option.c

```diff
@@ -969,10 +969,12 @@ void set_init_2(bool headless)
     p_window = Rows - 1;
   }
   set_number_default("window", Rows - 1);
+#if 0 // This bodges around problems that should properly be fixed in the TUI layer.
   if (!headless && !os_term_is_nice()) {
```

Comment:
Why not adjust os_term_is_nice() in the meantime? Unless you are certain the exceptions there are all accounted for now.